src/main/java/net/sf/rails/ui/swing/ORUIManager.java

```diff
@@ -1428,9 +1428,13 @@ public void updateStatus(PossibleAction actionToComplete, boolean myTurn) {
 
             // TEMPORARY extra message about having no route
             for (BuyTrain bTrain : possibleActions.getType(BuyTrain.class)) {
-                if (bTrain.isForcedBuyIfNoRoute()) {
+                if (bTrain.isForcedBuyIfHasRoute()) {
                     b.append("<br><font color=\"red\">");
-                    b.append(LocalText.getText("MustBuyTrainIfNoRoute"));
+                    if (bTrain.isForcedBuyIfNoRoute()) {
```

Comment:
Isn't this check semantically contrary to the one above (i.e. `bTrain.isForcedBuyIfHasRoute()`)?

Reply:
No, see my other comment on this class.